### PR TITLE
Switch to just and uv (#168)

### LINKED ISSUE #168: switch to just and uv
We're switching from make to just across our repositories. We should switch crashstats-tools to just as well.

While doing that, we can also switch to uv for project environment maintenance. That simplifies project scaffolding and documentation for dealing with virtual environments.

### PR DESCRIPTION
**Pull request by willkg**
This switches project environment to just and uv. Instead of using make rules, we now have just rules.

Running "just" by itself will list rules and documentation. The README has an additional "For develoers" section with similar information.

To test:

1. run `just devenv` to build a developer environment which includes a Python virtual environment with tox and other things necessary to maintain the project installed
2. run `just lint`, `just docs`, `just format`, `just test` -- these are the primary maintenance rules
3. run `just clean` which will remove artifacts and state from the project

Fixes #168

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+@_default:
+    just --list
+
+# Build a development environment
+devenv:
+    uv sync --extra dev --refresh --upgrade
+
+# Run tests and linting
+test *args: devenv
+    tox {{args}}
```
Comment (biancadanforth) [inline justfile:10]: By following the README instructions, this threw an error for me using `python3 -m venv venv` > `source venv/bin/activate` as my virtual env:

> For developing crashstats-tools, clone the Git repository, create a virtual environment, and install crashstats-tools and dev dependencies with::
>    $ just devenv


I got a warning from `uv`:
```
warning: `VIRTUAL_ENV=/Users/bdanforth/Projects/venv` does not match the project environment path `.venv` and will be ignored
```

After [looking this error up](https://github.com/astral-sh/uv/issues/7778), I ultimately had to set `UV_PROJECT_ENVIRONMENT=$VIRTUAL_ENV`. 

How are you setting your virtual env up? Perhaps this is worth mentioning?

**Review by biancadanforth — APPROVED** [review]
R+ though it might be worth mentioning a snag I hit in the README related to the virtual env and running the `just` recipes. See inline comment.

File: justfile (hunk 1 above)
Comment (willkg) [reply justfile:10]: Oh, whoops--I missed a `uv run` here. I'll fix that.

**Comment by willkg** [comment]
@biancadanforth ^^^ Does that fix the issue for you or help in any way?

**Comment by biancadanforth** [comment]
> @biancadanforth ^^^ Does that fix the issue for you or help in any way?

Yes this works now without adding any new env vars to my environment -- thanks!

I was looking into why, and from `uv help run`:
> When used in a project, the project environment will be created and updated before invoking the command.

I guess that'd take care of ensuring all the dependencies are installed in the directory `uv` expects.

**Comment by willkg** [comment]
Thank you!

**Pull request merged**